packages/identity-service/src/utils/waitForUser.ts

```diff
@@ -0,0 +1,52 @@
+import type { AudiusLibs } from '@audius/sdk'
+import type Logger from 'bunyan'
+
+export const waitForUser = async ({
+  userId,
+  handle,
+  blockNumber: blocknumber,
+  audiusLibsInstance,
+  logger
+}: {
+  userId: number
+  handle: string
+  blockNumber: number
+  audiusLibsInstance: AudiusLibs
+  logger: Logger
+}) => {
+  let retryCount = 0
+  while (retryCount < 10) {
```

Comment:
Will approve, but I think rather than using private makeRequest directly, we could hit the health check through proper means until the block number has passed and then grab the user too.

Reply:
how do you hit health check "through proper means"? Do we have a pattern for this?

Reply:
`audiusLibs.Users.getUsers` looked so promising with it's `minBlockNumber` but that's a totally different concept 🤦

Reply:
yuck. i guess we don't. i would advocate for cross-fetch with the selected discovery node and hitting the full users endpoint. just so that we don't have to re-migrate this later, but 🤷 dealer's choice

Reply:
I also tried hitting the full user endpoint, but when it 404s there's no block number info :joy:

Reply:
omg

Reply:
oh you know what, maybe i did the _non-full_ one?

Reply:
https://discoveryprovider.audius.co/v1/full/users/handle/sjoeijo3ij1ndoifsn no u right